src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventNotificationEmailManager.java

```diff
@@ -0,0 +1,175 @@
+                            .map(DetailedEventBookingDTO::getUserBooked)
+                            .map(UserSummaryDTO::getId)
+                            .distinct().collect(Collectors.toList());
+        for (Long id : ids) {
+            try {
+                RegisteredUserDTO user = userAccountManager.getUserDTOById(id);
+                emailManager.sendTemplatedEmailToUser(user,
+                    emailManager.getEmailTemplateDTO(templateId),
+                    new ImmutableMap.Builder<String, Object>()
+                        .put("event.emailEventDetails", event.getEmailEventDetails() == null ? "" : event.getEmailEventDetails())
+                        .put("event", event)
+                        .build(),
+                    EmailType.SYSTEM);
+                log.debug(String.format("Sent email to user: %s %s, at: %s", user.getGivenName(), user.getFamilyName(), user.getEmail()));
+            } catch (NoUserException e) {
+                log.error(String.format("No user found with ID: %s", id));
+            } catch (ContentManagerException e) {
+                log.error("Failed to add the scheduled email sent time: ", e);
+            }
+        }
+    }
+
+    public void sendReminderEmails() {
+        // Magic number
+        Integer limit = 10000;
+        Integer startIndex = 0;
+        Map<String, List<String>> fieldsToMatch = Maps.newHashMap();
+        Map<String, Constants.SortOrder> sortInstructions = Maps.newHashMap();
+        Map<String, AbstractFilterInstruction> filterInstructions = Maps.newHashMap();
+        fieldsToMatch.put(TYPE_FIELDNAME, Collections.singletonList(EVENT_TYPE));
+        sortInstructions.put(DATE_FIELDNAME, Constants.SortOrder.DESC);
+        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime threeDaysAhead = now.plusDays(3);
+        DateRangeFilterInstruction
+            eventsWithinThreeDays = new DateRangeFilterInstruction(new Date(), Date.from(threeDaysAhead.toInstant()));
+        filterInstructions.put(ENDDATE_FIELDNAME, eventsWithinThreeDays);
+
+        try {
+            ResultsWrapper<ContentDTO> findByFieldNames = this.contentManager.findByFieldNames(
+                properties.getProperty(CONTENT_INDEX), ContentService.generateDefaultFieldToMatch(fieldsToMatch),
+                startIndex, limit, sortInstructions, filterInstructions);
+            for (ContentDTO contentResult : findByFieldNames.getResults()) {
+                if (contentResult instanceof IsaacEventPageDTO) {
+                    IsaacEventPageDTO event = (IsaacEventPageDTO) contentResult;
+                    String emailKey = String.format("%s@pre", event.getId());
+                    List<BookingStatus> bookingStatuses = Arrays.asList(BookingStatus.CONFIRMED, BookingStatus.ATTENDED);
+                    if (!pgScheduledEmailManager.scheduledEmailAlreadySent(emailKey)) {
+                        pgScheduledEmailManager.saveScheduledEmailSent(emailKey);
```

Comment:
You'll probably be able to decide whether or not to act from the success of just the insert, but it's the sort of thing that's worth double checking in the documentation.

Reply:
The whole point of Quartz is that one and only one server will run each job, that's the only reason to use it and not just Cron. But you make a good point about races, since (hypothetically) the jobs could overlap.

I suppose `saveScheduledEmailSent` could return a boolean, based on whether the "insert on conflict no action" returned 1 rows or 0 rows added?

Reply:
Just using the `saveScheduledEmailSent` and having it return a boolean has been used